### PR TITLE
Make config replace cache

### PR DESCRIPTION
**Pull request by chadhietala**

### CODE REVIEW

File: lib/broccoli/broccoli-config-replace.js [hunk 1]
```diff
@@ -22,16 +24,54 @@ CustomReplace.prototype.build = function () {
...
 
-    this.processFile(config, filePath, destPath);
+    if (this._cache[key.hash]) {
+      this.writeFile(this._cache[key]);
+    } else {
+      filePath = path.join(inputPath, file);
+      destPath = path.join(this.outputPath, file);
+
+      this.processFile(config, filePath, destPath, key);
+    }
```
Comment (stefanpenner) [inline lib/broccoli/broccoli-config-replace.js]: we may want to evict cache entries

Comment (chadhietala) [inline lib/broccoli/broccoli-config-replace.js]: Yea it's doable. Would grow unbounded to the times you change the config.

**Comment by stefanpenner** [comment]
Although it seems like a very slow growing cache. If possible we should do our do diligence here.

**Comment by stefanpenner** [comment]
failing tests are unrelated, will look into them later this weekend

**Comment by stefanpenner** [comment]
we should extract this and write tests for it, and this PR is actually broken on rebuilds.

**Comment by stefanpenner** [comment]
fixed: https://github.com/ember-cli/ember-cli/commit/8ddf7b74e5eef247a9558caecf68ace2a1330ec0

**Pull request merged**